apps/cowswap-frontend/src/common/pure/ConfirmationPendingContent/ConfirmationPendingContent.tsx

```diff
@@ -1,68 +1,85 @@
-import { ReactNode } from 'react'
+import React, { ReactNode } from 'react'
+
+import { BackButton } from '@cowprotocol/ui'
+import { useWalletDisplayedAddress } from '@cowprotocol/wallet'
 
 import { Trans } from '@lingui/macro'
 import { CheckCircle, UserCheck } from 'react-feather'
 
 import { Wrapper } from './styled'
 import { UpperSection } from './styled'
-import { CloseIconWrapper } from './styled'
 import { WalletIcon } from './styled'
 import { LowerSection } from './styled'
 import { StepsWrapper } from './styled'
 import { StepsIconWrapper } from './styled'
 
+import { useWalletStatusIcon } from '../../hooks/useWalletStatusIcon'
+
 interface ConfirmationPendingContentProps {
   onDismiss: () => void
-  statusIcon: ReactNode
   title: string | ReactNode
   description: string | ReactNode
-  operationSubmittedMessage: string
   operationLabel: string
-  walletNameLabel?: string
-  walletAddress?: string
+  CustomBody?: ReactNode
+  CustomDescription?: ReactNode
 }
 
 export function ConfirmationPendingContent({
-  onDismiss,
-  statusIcon,
   title,
   description,
-  operationSubmittedMessage,
   operationLabel,
-  walletNameLabel = 'wallet',
-  walletAddress,
+  onDismiss,
+  CustomBody,
+  CustomDescription,
 }: ConfirmationPendingContentProps) {
+  const walletAddress = useWalletDisplayedAddress()
+  const statusIcon = useWalletStatusIcon()
+
+  const operationSubmittedMessage = `The ${operationLabel} is submitted.`
+
   return (
     <Wrapper>
       <UpperSection>
-        <CloseIconWrapper onClick={onDismiss} />
+        <BackButton onClick={onDismiss} />
         <WalletIcon>{statusIcon}</WalletIcon>
         <span>{title}</span>
       </UpperSection>
       <LowerSection>
         <h3>
-          <span>{description}</span>
+          <span>
+            {CustomDescription || (
+              <>
+                <span>{description} </span>
```

Comment:
Isn't `description` a custom description already?

Would it make sense to turn `description` type into `string|ReactNode` and use it the standard format here only when it's a string?

Reply:
Refactored it, thanks!

https://github.com/cowprotocol/cowswap/pull/3756/commits/1c13ea6724ca669627e4f47a49659982c3a2b4d0